### ISSUE TITLE
High DPI support

### ISSUE THREAD

**Issue by ccqil** [BoomingTech/Pilot#4]
My screen is 3840 x 2160 and everything so small that it's unreadable/unusable.
![](https://user-images.githubusercontent.com/24586659/161563355-e31e9cb5-1e2d-4b00-912e-e700ad2823b1.png)
This is a problem with imgui. And you can check [feature/hidpi-support](https://github.com/thedmd/imgui/tree/feature/hidpi-support) to fix it.

**Comment by JX-Master** [comment]
On Windows 10 1607 and above, the Win32 API [`GetDpiForWindow`](https://docs.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-getdpiforwindow) will fetch the DPI value for the specified window. The value is 96 by default, and the scaled UI size can be computed by:
```
int zoom = GetDpiForWindow(window->m_hwnd);
float scale = (float)zoom / 96.0f;
width = (unsigned int)(width / scale);
height = (unsigned int)(height / scale);
```
As far as I know, there is no equivalent API on Mac platforms.

**Comment by hybcloud** [comment]
just double the font size, almost everthing in imgui adjust to font size.

**Comment by ShenMian** [comment]
@ccqil 或许你可以尝试以下 #79 中的修改.

**Comment by iaomw** [comment]
> On Windows 10 1607 and above, the Win32 API [`GetDpiForWindow`](https://docs.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-getdpiforwindow) will fetch the DPI value for the specified window. The value is 96 by default, and the scaled UI size can be computed by:
> 
> ```
> int zoom = GetDpiForWindow(window->m_hwnd);
> float scale = (float)zoom / 96.0f;
> width = (unsigned int)(width / scale);
> height = (unsigned int)(height / scale);
> ```
> 
> As far as I know, there is no equivalent API on Mac platforms.

There are surely some native API dealing DPI on macOS. Apple has its ownAPIs almost for anything.
If we don't want introduce new libraries, ImGui and GLFW could also do the job. ImGui is buggy though test, so prefer GLFW.

<img width="955" alt="image" src="https://user-images.githubusercontent.com/2307692/162419513-0fb218ad-3451-4622-a3a0-f166e634025e.png">

**Comment by hyv1001** [comment]
Should be fixed by #85, feel free to reopen if issue still exists
